river/Seat.zig

```diff
@@ -405,6 +405,23 @@ pub fn handleMapping(
     return handled;
 }
 
+/// Handle any user-defined mapping for passed button and modifiers.
+/// Returns true if there was a mapping.
+pub fn handleButtonMapping(
+    self: *Self,
+    event_code: u32,
+    modifiers: wlr.Keyboard.ModifierMask,
+) bool {
+    const modes = &server.config.modes;
+    for (modes.items[self.mode_id].button_mappings.items) |*mapping| {
```

Comment:
```suggestion
    for (modes.items[self.mode_id].button_mappings.items) |mapping| {
```

You only need a pointer if you want to modify the item.

Reply:
Thank you!